Wabbajack.Networking.WabbajackClientApi/Client.cs

```diff
@@ -548,47 +550,67 @@ public async Task<IReadOnlyList<string>> GetMyModlists(CancellationToken token)
         return (await _dtos.DeserializeAsync<string[]>(await response.Content.ReadAsStreamAsync(token), token))!;
     }
 
-    public async Task PublishModlist(string namespacedName, Version version,  AbsolutePath modList, DownloadMetadata metadata)
+    public async Task<(IObservable<(Percent PercentDone, string Message)> Progress, Task PublishTask)> PublishModlist(
+    string namespacedName, Version version, AbsolutePath modList, DownloadMetadata metadata)
     {
         var pair = namespacedName.Split("/");
         var wjRepoName = pair[0];
         var machineUrl = pair[1];
 
         var repoUrl = (await LoadRepositories())[wjRepoName];
-
         var decomposed = repoUrl.LocalPath.Split("/");
         var owner = decomposed[1];
         var repoName = decomposed[2];
         var path = string.Join("/", decomposed[4..]);
-        
-        _logger.LogInformation("Uploading modlist {MachineUrl}", namespacedName);
-        
+
+        var uploadProgress = new double();
         var (progress, uploadTask) = await UploadAuthorFile(modList);
-        progress.Subscribe(x => _logger.LogInformation(x.Message));
-        var downloadUrl = await uploadTask;
         
-        _logger.LogInformation("Publishing modlist {MachineUrl}", namespacedName);
+        var disposable = progress.Subscribe(m =>
```

Comment:
Instead of doing this, I would take a look at the Subject used in UploadAuthorFile, that way you don't have to manually mess around with the observable and a subscribe